hnn_core/network_builder.py

```diff
@@ -575,47 +558,30 @@ def _parnet_connect(self):
         self._connect_celltypes('L2Pyr', target_cell, 'soma', 'ampa',
                                 nc_dict)
 
-        # common feed -> xx
-        for p_common in self.net._p_common:
-            for target_cell_type in ['L2Basket', 'L5Basket', 'L5Pyr', 'L2Pyr']:
-                if (target_cell_type == 'L5Basket' and
-                        p_common['loc'] == 'distal'):
-                    continue
-                for receptor in ['ampa', 'nmda']:
-                    if f'{target_cell_type}_{receptor}' in p_common.keys():
-                        nc_dict['lamtha'] = p_common['lamtha']
-                        nc_dict['A_weight'] = \
-                            p_common[f'{target_cell_type}_{receptor}'][0]
-                        nc_dict['A_delay'] = \
-                            p_common[f'{target_cell_type}_{receptor}'][1]
-                        self._connect_celltypes('common', target_cell_type,
-                                                p_common['loc'], receptor,
-                                                nc_dict)
-
-        # unique feed -> xx
-        p_unique = self.net._p_unique
-        for src_cell_type in p_unique:
-
-            p_src = p_unique[src_cell_type]
+        # loop over _all_ drives, _connect_celltypes picks ones on this rank
+        for drive in self.net.external_drives.values():
+
             receptors = ['ampa', 'nmda']
-            if src_cell_type == 'extgauss':
+            if drive['type'] == 'gaussian':
                 receptors = ['ampa']
-
-            for target_cell_type in ['L2Basket', 'L5Basket', 'L5Pyr', 'L2Pyr']:
-                # XXX: hack for distal connection
-                if target_cell_type == 'L5Basket' and p_src['loc'] == 'distal':
+            # conn-parameters are for each target cell type
+            for target_cell_type, drive_conn in drive['conn'].items():
+                # XXX: hack for distal connection XXX: Why?!
```

Comment:
you would get a key error down below because the dictionaries returned by `create_pext` would not contain the key for distal. You can try removing this in `master` and see what I mean. Maybe it works now with your changes?

Reply:
removed in fdb4fa8 (tests pass fine)